packages/plugins/typescript-react-apollo/src/index.ts

```diff
@@ -42,10 +42,16 @@ export const plugin: PluginFunction<TypeScriptReactApolloConfig> = async (
 export const validate: PluginValidateFn<any> = async (
   schema: GraphQLSchema,
   documents: DocumentFile[],
-  config: any,
+  config: TypeScriptReactApolloConfig,
   outputFile: string
 ) => {
-  if (extname(outputFile) !== '.tsx') {
-    throw new Error(`Plugin "react-apollo" requires extension to be ".tsx"!`);
+  if (config.noComponents) {
+    if (extname(outputFile) !== '.ts') {
```

Comment:
Mh... why should we force `ts` instead of `tsx`? I know that in a ts file you can't use pragma directives, but banning it seems like.. too much, don't you think so?
Thansk!

Reply:
You're right fixed :)